### ISSUE TITLE
Compute F1 score for multilabel classifier

### ISSUE THREAD

**Issue by MHDBST** [tensorflow/tensorflow#27171]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.13
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
I am working with tf.contrib.metrics.f1_score in a metric function and call it using an estimator. I want to compute the F1 score for **multi label classifier** but this contrib function can not compute it. Please add this capability to this F1 ( computing macro and micro f1). I need it to compare the dev set and based on that keep the best model. I don't like to compute it using the sklearn

**Will this change the current api? How?**
IDK

**Who will benefit with this feature?**
Every one who is trying to compute macro and micro f1 inside the Tensorflow function and not willing to use other python libraries. 

**Any Other info.**
If it is possible to compute macro f1 score in tensorflow using tf.contrib.metrics please let me know.

Thanks

**Comment by shashvatshahi1998** [comment]
@alextp there is no function like f1_score in tf.keras.metrics it is only in tf.contrib so where can we add functions for macros and micros, can you please guide me a little bit.

**Comment by alextp** [comment]
Maybe this belongs in some other package like tensorflow/addons or tf-text?

**Comment by MHDBST** [comment]
Where can we find macro f1 function? Is it anywhere in tf. ANYWHERE?! Is it developed or added or not?

**Comment by ymodak** [comment]
@MHDBST  f1_score is defined in https://github.com/tensorflow/tensorflow/blob/0fb46cf9139f1e8cf88195e0b8f555eb10d00b3b/tensorflow/contrib/metrics/python/metrics/classification.py#L70 However there is no feature to compute macro f1 score. PR #27171 addresses this feature. Thanks!

**Comment by MHDBST** [comment]
Thanks @ymodak, this f1 function is not working for multiclass classification ( more than two labels). Is there any way to compute F1 for multi class classification?

**Comment by shashvatshahi1998** [comment]
I will work on that surely

**Comment by shashvatshahi1998** [comment]
I will work on that

**Comment by ymodak** [comment]
@MHDBST As a workaround, have you explored https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html. This can help you compute f1_score for binary as well as multi-class classification problems. Thanks!

**Comment by MHDBST** [comment]
@ymodak This function is what I'm using now. But I think there should be a metric in Tensorflow like accuracy, or F1 ( for binary classification) to compute macro f1 (for multi class classification) independent from other libraries.
